src/BuiltInTools/dotnet-watch/Internal/ProcessRunner.cs

```diff
@@ -109,9 +111,34 @@ private Process CreateProcess(ProcessSpec processSpec)
                 process.StartInfo.Environment.Add(env.Key, env.Value);
             }
 
+            SetEnvironmentVariable(process.StartInfo, "DOTNET_STARTUP_HOOKS", processSpec.EnvironmentVariables.DotNetStartupHooks, Path.PathSeparator);
+            SetEnvironmentVariable(process.StartInfo, "ASPNETCORE_HOSTINGSTARTUPASSEMBLIES", processSpec.EnvironmentVariables.AspNetCoreHostingStartupAssemblies, ';');
+
             return process;
         }
 
+        private void SetEnvironmentVariable(ProcessStartInfo processStartInfo, string envVarName, List<string> envVarValues, char separator)
+        {
+            if (envVarValues is { Count: 0 })
```

Comment:
These C# 8 and 9 features are interesting...

Why is this preferred over `envVarValues.Count == 0`?

Reply:
This includes a null check which I don't need, but I think I was feeling fancy:

```
if (envVarValues != null && envVarVules.Count == 0)
```

Reply:
Ah, that is actually somewhat better then.